libsolidity/ast/Types.cpp

```diff
@@ -561,6 +812,75 @@ shared_ptr<IntegerType const> IntegerConstantType::integerType() const
 		);
 }
 
+shared_ptr<FixedPointType const> RationalNumberType::fixedPointType() const
+{
+	//do calculations up here
```

Comment:
I did not say that it is perfect, but at least to me, it looks a lot simpler. I might be missing out on some edge cases, but actually I think I got all of them.